src/content-helper/dashboard-widget/top-posts/component-list.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import { Spinner } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import DashboardWidgetProvider from '../provider';
+import TopPostListItem from './component-list-item';
+import { TopPostData } from './model';
+import { ContentHelperError } from '../../../blocks/content-helper/content-helper-error';
+import { getDateInUserLang, SHORT_DATE_FORMAT } from '../../../blocks/shared/utils/date';
+
+const FETCH_RETRIES = 3;
+
+/**
+ * List of the top posts.
+ */
+function TopPostList() {
+	const [ loading, setLoading ] = useState<boolean>( true );
+	const [ error, setError ] = useState<ContentHelperError>();
+	const [ posts, setPosts ] = useState<TopPostData[]>( [] );
+	const provider = new DashboardWidgetProvider();
+
+	useEffect( () => {
+		const fetchPosts = async ( retries: number ) => {
+			provider.getTopPosts()
+				.then( ( result ): void => {
```

Comment:
Can we add type on param `result`?

Reply:
We're doing the same thing in the Related Top Posts list. Maybe this would go into the refactorings. What would you suggest doing here exactly?

Reply:
Only suggesting to add typing which can be helpful while understanding the code.